lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -277,9 +265,29 @@ void StagesSubject::stageChanged(UsdNotice::ObjectsChanged const& notice, UsdSta
 	}
 }
 
-void StagesSubject::onStageSet(const UsdMayaProxyStageSetNotice& notice)
+void StagesSubject::onStageSet(const MayaUsdProxyStageSetNotice& notice)
+{
+	// We should have no listerners and stage map is dirty.
+	TF_VERIFY(g_StageMap.isDirty());
+	TF_VERIFY(fStageListeners.empty());
+
+	StagesSubject::Ptr me(this);
+	for (auto stage : ProxyShapeHandler::getAllStages())
+	{
+		fStageListeners[stage] = TfNotice::Register(
+			me, &StagesSubject::stageChanged, stage);
+	}
```

Comment:
When we get the stage set notice is when we can register. The stage(s) is(are) valid at this point.